src/main/java/com/commercetools/sync/sdk2/types/utils/FieldDefinitionsUpdateActionUtils.java

```diff
@@ -0,0 +1,288 @@
+                                TypeRemoveFieldDefinitionActionBuilder.of()
+                                    .fieldName(oldFieldDefinitionName)
+                                    .build(),
+                                TypeAddFieldDefinitionActionBuilder.of()
+                                    .fieldDefinition(newFieldDefinition)
+                                    .build());
+                          }
+                        } else {
+                          return new ArrayList<TypeUpdateAction>();
+                        }
+                      })
+                  .orElseGet(
+                      () ->
+                          singletonList(
+                              TypeRemoveFieldDefinitionActionBuilder.of()
+                                  .fieldName(oldFieldDefinitionName)
+                                  .build()));
+            })
+        .flatMap(Collection::stream)
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Compares the field types of the {@code fieldTypeA} and the {@code fieldTypeB}.
+   *
+   * @param fieldTypeA the first type to compare.
+   * @param fieldTypeB the second field type to compare.
+   * @return true if both field types equal, false otherwise.
+   */
+  private static boolean haveSameFieldType(
+      @Nonnull final FieldType fieldTypeA, @Nonnull final FieldType fieldTypeB) {
+    return fieldTypeA.getClass() == fieldTypeB.getClass();
+  }
+
+  /**
+   * Compares the order of a list of old {@link FieldDefinition}s and a list of new {@link
+   * FieldDefinition}s. If there is a change in order, then a change field definition order (with
+   * the new order) is built. If there are no changes in order an empty optional is returned.
+   *
+   * @param oldFieldDefinitions the list of old {@link FieldDefinition}s
+   * @param newFieldDefinitions the list of new {@link FieldDefinition}s
+   * @return a list of field definition update actions if the order of field definitions is not
+   *     identical. Otherwise, if the field definitions order is identical, an empty optional is
+   *     returned.
+   */
+  @Nonnull
+  private static Optional<TypeUpdateAction> buildChangeFieldDefinitionOrderUpdateAction(
+      @Nonnull final List<FieldDefinition> oldFieldDefinitions,
+      @Nonnull final List<FieldDefinition> newFieldDefinitions) {
+
+    final List<String> newNames =
+        newFieldDefinitions.stream().map(FieldDefinition::getName).collect(toList());
+
+    final List<String> existingNames =
+        oldFieldDefinitions.stream()
+            .map(FieldDefinition::getName)
+            .filter(newNames::contains)
+            .collect(toList());
+
+    final List<String> notExistingNames =
+        newNames.stream().filter(newName -> !existingNames.contains(newName)).collect(toList());
+
+    final List<String> newFieldDefinitionsOrderNames =
+        newFieldDefinitions.stream().map(FieldDefinition::getName).collect(toList());
```

Comment:
Why this? Does `.filter` change the original `newNames` list?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/1007/commits/a5336d635f961199f1ba70d85ec3e417591f96a8